Terminal.Gui/ConsoleDrivers/NetDriver.cs

```diff
@@ -1092,11 +1112,50 @@ Key MapKey (ConsoleKeyInfo keyInfo)
 
 			return (Key)((uint)Key.F1 + delta);
 		}
-		if (keyInfo.KeyChar != 0) {
-			return MapKeyModifiers (keyInfo, (Key)((uint)keyInfo.KeyChar));
+
+		// Is it a key between a..z?
+		if ((char)keyInfo.KeyChar is >= 'a' and <= 'z') {
+			// 'a' should be Key.A
+			return (Key)((uint)keyInfo.KeyChar) & ~Key.Space;
 		}
 
-		return (Key)(0xffffffff);
+		// Is it a key between A..Z?
+		if (((Key)((uint)keyInfo.KeyChar) & ~Key.Space) is >= Key.A and <= Key.Z) {
+			// It's Key.A...Z.  Make it Key.A | Key.ShiftMask
+			return (Key)((uint)keyInfo.KeyChar) & ~Key.Space | Key.ShiftMask;
+		}
+
+		return (Key)(uint)keyInfo.KeyChar;
+	}
+
+	/// <summary>
+	/// Identifies the state of the "shift"-keys within a event.
+	/// </summary>
+	public class KeyModifiers {
```

Comment:
You have got rid the `KeyModifiers`. Why recreating here again?

Reply:
Because I'm extremely tired of working on this PR and NetDriver confuses the hell out of me. ;-).

I should have added a comment saying this is temporary.

Reply:
@tig I submitted the PR https://github.com/tig/Terminal.Gui/pull/18 to your branch that fix the NetDriver.